### PR TITLE
Prepare to release v1.0.0

### PR DESCRIPTION
**Pull request by jeremyfowers**
This PR:
- Sets the package version number to v1.0.0
- Addresses #55 by adding the public APIs list to the contributions guide (see #62 and #63 for related future works).
- Addresses #57 by adding a release notes doc with an intro, v1.0.0 release notes, and template for future release notes.
- Moves the `logged_subprocess()` function from the `common.build` module to `run.plugin_helpers`, since it is never used during build and is only used during runtime plugins.

### CODE REVIEW

**Comment by jeremyfowers** [comment]
Release candidate available for testing: https://test.pypi.org/project/turnkeyml/1.0.0/

File: docs/release_notes.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+If you are creating the release notes for a new version, please see the [template](#template-version-majorminorpatch). Release notes should capture all of the significant changes since the last numbered package release.
+
+# Version 1.0.0
+
+This version focuses on cleaning up technical debts. It removes cumbersome requirements and unused features, while also clarifying some naming schemes.
+
+## Improvements
+
+### CLI Users
```
Comment (jeremyfowers) [inline docs/release_notes.md:17]: Remove "CLI" and "API"

File: docs/release_notes.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+# Release Notes
+
+This document tracks the major changes in each package release of TurnkeyML.
+
+We are tracking two types of major changes:
+ - New features that enhance the user and developer experience
+ - Breaking changes to the CLI or public APIs
```
Comment (jeremyfowers) [inline docs/release_notes.md:7]: Refactor: Users vs. Developers are the top-level sections

File: docs/release_notes.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+ - The `quantization_samples` argument to the `build_model()` API has been removed.
+ - The naming scheme of the members of `Stats` has been adjusted for consistency. It used to refer to both builds and benchmarks as "builds", whereas now it uses "evaluations" as a superset of the two.
+   - `Stats.add_build_stat()` is now `Stats.save_model_eval_stat()`.
+   - `Stats.add_build_sub_stat()` is now `Stats.save_model_eval_sub_stat()`.
+   - `Stats.stat_id` is now `Stats.evaluation_id`.
+   - The `builds` section of the stats/reports is now `evaluations`.
+   - `Stats.save_stat()` is now `Stats.save_model_stat()`.
+   - `Stats.build_stats` is now `Stats.evaluation_stats`.
+ - The `SetSuccess` build stage has been removed because build success has been reworked (see improvements).
+ - The `logged_subprocess()` API has been moved from the `common.build` module to the `run.plugin_helpers` module.
```
Comment (danielholanda) [inline docs/release_notes.md:50]: Should we also add those to the "Describe this release" field when creating the actual release?
https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository

**Review by danielholanda — APPROVED** [review]
Looks good!

File: docs/release_notes.md (hunk 3 above)
Comment (jeremyfowers) [reply docs/release_notes.md:50]: We aren't managing releases in any formal way (according to GitHub) right now. But something to look into in the future!

File: docs/release_notes.md (hunk 2 above)
Comment (jeremyfowers) [reply docs/release_notes.md:7]: done

File: docs/release_notes.md (hunk 1 above)
Comment (jeremyfowers) [reply docs/release_notes.md]: done

**Pull request merged**